### PR TITLE
Add clone method to Reply

### PR DESCRIPTION
**Pull request by ahcorde**
Add clone method to Reply

Required here https://github.com/ros2/rmw_zenoh/pull/311/files

### CODE REVIEW

File: include/zenoh/api/reply.hxx [hunk 1]
```diff
@@ -84,6 +84,16 @@ class Reply : public Owned<::z_owned_reply_t> {
         return {};
     }
 #endif
+
+    /// @brief Construct a a shallow copy of this Reply.
+    /// This can be used, to send reply replies outside of ``Queryable`` callback.
```
Comment (DenisBiryukov91) [inline include/zenoh/api/reply.hxx:89]: the documentation is wrong, since it is describes the behavior of Query clone.

Comment (ahcorde) [reply include/zenoh/api/reply.hxx:89]: Can you help me with right documentation here ?

Comment (DenisBiryukov91) [reply include/zenoh/api/reply.hxx:89]: I think  just `/// @brief Construct a a shallow copy of this Reply.` should be good enough.

Comment (DenisBiryukov91) [reply include/zenoh/api/reply.hxx]: Sorry, could you also remove an extra `a` before `shallow_copy` ?

File: include/zenoh/api/reply.hxx [hunk 2]
```diff
@@ -84,7 +84,16 @@ class Reply : public Owned<::z_owned_reply_t> {
         return {};
     }
 #endif
+
+    /// @brief Construct a shallow copy of this Reply.
+    ///
+    /// The reply will be sent only when the last clone is destroyed.
+    Reply clone() const {
+        Reply q(zenoh::detail::null_object);
```
Comment (milyin) [inline include/zenoh/api/reply.hxx:92]: Could you please also rename this "q" variable to something meaningful? E.g. `Reply reply` or something like that

**Review by milyin — CHANGES_REQUESTED** [review]

**Review by milyin — APPROVED** [review]

**Pull request merged**